tests/unit/src/test/scala/org/scalafmt/tests/TermSuite.scala

```diff
@@ -46,8 +46,8 @@ object TermSuite extends BaseScalaPrinterTest {
           |  (<b></b>)
           |}
         """.stripMargin)
-  check("(a + b) { c => d }")
-  check("(a + b) { case c => d }")
+  check("((a + b)) { c => d }")
```

Comment:
Why are these parentheses needed? The trees are the same
```scala
ollie-scala@ q"(a + b) { _ + 1 }"
res1: Term.Apply = Term.Apply(
  Term.ApplyInfix(Term.Name("a"), Term.Name("+"), List(), List(Term.Name("b"))),
  List(Term.Block(List(Term.ApplyInfix(_, Term.Name("+"), List(), List(Lit.Int(1))))))
)

ollie-scala@ q"((a + b)) { _ + 1 }"
res2: Term.Apply = Term.Apply(
  Term.ApplyInfix(Term.Name("a"), Term.Name("+"), List(), List(Term.Name("b"))),
  List(Term.Block(List(Term.ApplyInfix(_, Term.Name("+"), List(), List(Lit.Int(1))))))
)
```

Reply:
check also does the diff, Do you prefer if we use `checkStructural` here ?

Reply:
It seems our precedence groups could be stricter to avoid that parenthesis.

Reply:
I think we should special case when Term.New is the qualifier in Term.Apply.

Reply:
There was a miscommunication we cleared out offline: @olafurpg wants correctness but also does not want to add extra parenthesis when not necessary.